packages/flame/lib/src/text/sprite_font_renderer.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'package:flame/src/anchor.dart';
+import 'package:flame/src/text/text_renderer.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+/// [TextRenderer] implementation that uses a spritesheet of various font glyphs
+/// to render text.
+///
+/// For example, suppose there is a spritesheet with sprites for characters from
+/// A to Z. Mapping these sprites into a [SpriteFontRenderer] allows then to
+/// write text using these sprite images as the font.
+///
+/// Currently, this class supports monospace fonts only -- the width and the
+/// height of all characters must be the same ([charWidth] and [charHeight]).
+/// Extra space between letters can be added via the [letterSpacing] parameter
+/// (it can also be negative to "squish" characters together). Finally, the
+/// [scale] parameter allows scaling the font to be bigger/smaller relative to
+/// its size in the source image.
+///
+/// The [paint] parameter is used to composite the character images onto the
+/// canvas. Its default value will draw the character images as-is. Changing
+/// the opacity of the paint's color will make the text semi-transparent.
+class SpriteFontRenderer extends TextRenderer {
+  SpriteFontRenderer({
+    required this.source,
+    required this.charWidth,
+    required this.charHeight,
+    this.letterSpacing = 0,
+    this.scale = 1,
+  });
+
+  final Image source;
+  final double charWidth;
+  final double charHeight;
+  final double scale;
+  double letterSpacing;
+  bool get isMonospace => true;
+
+  Map<int, _GlyphInfo> glyphs = {};
```

Comment:
Can we make this private since it is internal and the value is an internal class?

Reply:
done